lib/_http_server.js

```diff
@@ -823,7 +823,11 @@ function socketOnError(e) {
   }
 
   if (!this.server.emit('clientError', e, this)) {
-    if (this.writable && this.bytesWritten === 0) {
+    // Cautions must be taken to avoid corrupting the remote peer,
+    // reply an error segment if no in-flight ServerResponse
+    // or data of the in-flight one is not yet written to this socket
```

Comment:
```suggestion
    // Caution must be taken to avoid corrupting the remote peer.
    // Reply an error segment if there is no in-flight `ServerResponse`,
    // or no data of the in-flight one has been written yet to this socket.
```